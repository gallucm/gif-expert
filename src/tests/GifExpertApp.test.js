const { shallow } = require("enzyme");
const { default: GifExpertApp } = require("../GifExpertApp");

describe('Pruebas en <GifExpertApp>', () => {
    
    const wrapper = shallow(<GifExpertApp/>);   

    test('Debe poder cargarse la snapshot correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe mostrar una categoria', () => {
        const category = 'Dragon Ball';

        let wrapper = shallow(<GifExpertApp defaultCategory={category}/>);

        expect(wrapper).toMatchSnapshot();

        expect(wrapper.find('GifGrid').length).toEqual(1);
    })
    
    
})
