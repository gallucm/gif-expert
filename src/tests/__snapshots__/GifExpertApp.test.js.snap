// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp> Debe mostrar una categoria 1`] = `
<Fragment>
  <h2>
    Buscador de GIFS
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ul>
    <GifGrid
      category="Dragon Ball"
    />
  </ul>
</Fragment>
`;

exports[`Pruebas en <GifExpertApp> Debe poder cargarse la snapshot correctamente 1`] = `
<Fragment>
  <h2>
    Buscador de GIFS
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ul>
    <GifGrid
      category=""
    />
  </ul>
</Fragment>
`;
